### ISSUE TITLE
Bug after `fix data leakage`

### ISSUE THREAD

**Issue by zhouzq-thu** [xtcpete/rdd#23]
## Calling `softdetect` with `normalized_coordinates=False`,

https://github.com/xtcpete/rdd/blob/cacd40e61947d08d8bb77a02dc573e94113411c5/src/RDD.py#L89

https://github.com/xtcpete/rdd/blob/cacd40e61947d08d8bb77a02dc573e94113411c5/src/RDD.py#L158

## leads the grid is not normalized in grid_sample.

https://github.com/xtcpete/rdd/blob/cacd40e61947d08d8bb77a02dc573e94113411c5/src/models/soft_detect.py#L126-L130

## Possible Fix
```python
            # compute result keypoints
            keypoints_xy_nms = torch.stack([indices_kpt % w, indices_kpt // w], dim=1)  # Mx2
            keypoints_xy = keypoints_xy_nms + xy_residual
            keypoints_xy_norm = keypoints_xy / keypoints_xy.new_tensor([w - 1, h - 1]) * 2 - 1  # (w,h) -> (-1~1,-1~1)

            kptscore = torch.nn.functional.grid_sample(scores_map[b_idx].unsqueeze(0), keypoints_xy_norm.view(1, 1, -1, 2),
                                                       mode='bilinear', align_corners=True)[0, 0, 0, :]  # CxN

            keypoints.append(keypoints_xy_norm if normalized_coordinates else keypoints_xy)
            scoredispersitys.append(scoredispersity)
            kptscores.append(kptscore)
```

**Comment by xtcpete** [comment]
Hi, thanks for bringing this up. This is intended as the interpolator will do normalization.
